File: sdks/java/src/main/java/com/puxbay/resources/StocktakesResource.java
package com.puxbay.resources;

import com.puxbay.Puxbay;
import com.puxbay.exceptions.PuxbayException;
import com.puxbay.models.PaginatedResponse;
import com.puxbay.models.Stocktake;

public class StocktakesResource {
    private final Puxbay client;

    public StocktakesResource(Puxbay client) {
        this.client = client;
    }

    public PaginatedResponse<Stocktake> list(int page) throws PuxbayException {
        return (PaginatedResponse<Stocktake>) client.request("GET", "stocktakes/?page=" + page, null, PaginatedResponse.class);
    }

    public Stocktake get(String stocktakeId) throws PuxbayException {
        return client.request("GET", "stocktakes/" + stocktakeId + "/", null, Stocktake.class);
    }

    public Stocktake create(Stocktake stocktake) throws PuxbayException {
        return client.request("POST", "stocktakes/", stocktake, Stocktake.class);
    }

    public Stocktake complete(String stocktakeId) throws PuxbayException {
        return client.request("POST", "stocktakes/" + stocktakeId + "/complete/", null, Stocktake.class);
    }
}

File: sdks/kotlin/src/main/kotlin/com/puxbay/resources/StocktakesResource.kt
package com.puxbay.resources

import com.puxbay.Puxbay
import com.puxbay.models.Stocktake
import com.puxbay.models.PaginatedResponse
import io.ktor.http.HttpMethod

class StocktakesResource(client: Puxbay) : BaseResource(client) {
    suspend fun list(page: Int = 1): PaginatedResponse<Stocktake> =
        client.request(HttpMethod.Get, "stocktakes/?page=$page")
    
    suspend fun get(stocktakeId: String): Stocktake =
        client.request(HttpMethod.Get, "stocktakes/$stocktakeId/")
    
    suspend fun create(stocktake: Stocktake): Stocktake =
        client.request(HttpMethod.Post, "stocktakes/", stocktake)
    
    suspend fun complete(stocktakeId: String): Stocktake =
        client.request(HttpMethod.Post, "stocktakes/$stocktakeId/complete/")
}
